.pre-commit-config.yaml

```diff
@@ -31,6 +31,11 @@ repos:
             'flake8-rst-docstrings>=0.2.3',
             'pydocstyle>=6.0.0',
         ]
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.4.1
+    hooks:
+    -   id: mypy
+        files: '^Bio.*'
```

Comment:
This looks odd... we have no files named ``Bio.*`` where dot is literal. We do have ``Bio/*`` though (and ``BioSQL/*``)

Reply:
This is a regular expression

Reply:
I presume you want to match (Python) files under ``Bio/`` and ``BioSQL/``

The caret to left-anchor the search makes sense.

Why use dot (any char) and star (zero or more chars)?

Reply:
> I presume you want to match (Python) files under `Bio/` and `BioSQL/`

Yes correct. This essentially excludes `Tests` and `Scripts` for now.

It should match the `files` selection in [`.mypy.ini`](https://github.com/speleo3/biopython/blob/mypy/.mypy.ini#L2) which is used when you run `mypy` without any path arguments (not via `pre-commit`, which will pass paths of the committed files).

> Why use dot (any char) and star (zero or more chars)?

The `.*` is indeed redundant, I can remove it. Now that you pointed it out, I would actually prefer a more explicit pattern like `'^Bio/|^BioSQL/'` or `'^(Bio|BioSQL)/'`.

Reply:
I think ``'^(Bio|BioSQL)/'`` is much clearer too :smile: